tiflash/tiflash-command-line-flags.md

```diff
@@ -13,3 +13,69 @@ This document introduces the command-line flags that you can use when you launch
+
+- User scenarios:
+
+    - If you need to downgrade TiFlash from a version >= v5.4.0 that has enabled data validation to a version < v5.4.0, you can use this tool to downgrade the data format of the DTFile.
+    - If you upgrade TiFlash to a version >= v5.4.0, and if you hope to enable data validation for existing data, you can use this tool to upgrade the data format of the DTFile.
+    - Test the space usage and read speed of the DTFile in different configurations.
+
+- Parameters:
+    - `--imitative`: When you do not use the encryption feature of the DTFile, you can use this flag to avoid using the configuration file and connecting to PD.
+    - `--version`: The version of DTFile. The value options are `1` and `2` (default). `1` is the old version, and `2` is the version corresponding to the new checksum.
+    - `--algorithm`: The hash algorithm used for data validation. The value options are `xxh3` (default), `city128`, `crc32`, `crc64`, and `none`. This parameter is effective only when `version` is `2`.
+    - `--frame`: The size of the validation frame. The default value is `1048576`. This parameter is effective only when `version` is `2`.
+    - `--compression`: The target compression algorithm. The value options are `lz4` (default), `lz4hc`, `zstd`, and `none`.
+    - `--level`: The target compression level. The default value is `-1`, which means automatic mode. The value range varies with different compression algorithms.
+    - `--config-file`: The configuration file of `dttool migrate` is the same as the [configuration file of `server`](/tiflash/tiflash-command-line-flags.md#server---config-file). When you use the configuration file, exit the local TiFlash server instance. For more information, see `--imitative`.
+    - `--file-id`: The ID of the DTFile. For example, the ID of the DTFile `dmf_123` is `123`.
+    - `--workdir`: The parent directory of `dmf_xxx`.
+    - `--dry`: The dry run mode. Only the migration process is output.
+    - `--nokeep`: Does not keep the original data. When this option is not enabled, `dmf_xxx.old` files are created.
+
+> **Warning:**
+>
+> TiFlash can read DTFile that uses custom compression algorithms and compression levels. However, only the `lz4` algorithm of the default compression level is officially supported. Custom compression parameters have not been thoroughly tested and are only experimental.
+
+> **Note:**
+>
+> For security reasons, DTTool attempts to add lock to the working directory in the migration mode. Therefore, in the same directory at the same time, only one DTTool can perform a migration task. If you forcibly stop DTTool where the lock is not released, when you try to rerun DTTool later, DTTool might refuse to perform the migration task.
```

Comment:
```suggestion
> For security reasons, DTTool attempts to add a lock to the working directory in the migration mode. Therefore, in the same directory, only one DTTool can perform the migration task at the same time. If you forcibly stop DTTool where the lock is not released, then when you try to rerun DTTool later, it might refuse to perform the migration task.
```